### PR TITLE
agenda: add php 8 discussion

### PR DESCRIPTION
**Pull request by boesing**
Feedback is very welcome. 
Should I add/change something?

### CODE REVIEW

File: meetings/agenda.md [hunk 1]
```diff
@@ -7,3 +7,20 @@ Please file pull requests to add, or discuss items to add, to the agenda.
 
 ## Items to discuss
 
+- PHP 8.0 introduction in packages
+
+  As we are about to support PHP 8.0, what is our strategy to do so?
+  - Do we want to support PHP `8.0` or `8.*`? (`php: ^8.0` *vs* `php: ~8.0.0`)
```
Comment (samsonasik) [inline meetings/agenda.md:13]: probably mention keep current minimal version?

```json
php: ^7.1 || ^8.0 
```

or 

```json
php: ^5.6 || ^7.0 || ^8.0 
```

Comment (weierophinney) [reply meetings/agenda.md:13]: I think if we add support for 8, this will be in a minor version, at which point we can drop support for 5.6.

Comment (michalbundyra) [reply meetings/agenda.md:13]: I'd say not only PHP 5.6 but everything what is < 7.3 (PHP 7.2 is "alive" for 3 more months only)

**Comment by weierophinney** [comment]
I think this is great!

HOWEVER

MAKE IT A PROPOSAL!

Tell us how YOU think it should be done. This will make for something concrete to discuss. I've found that when we have open-ended items on the agenda, discussion generally ends up going around in circles. Having something concrete is good, as if somebody feels strongly against it or some part of it, we have a more targeted discussion that can be resolved during the meeting.

So, do something along the lines of:

> PHP 8 is coming out in a few months, and we need a strategy for preparing our packages for it. I propose:
> 
> - We use `~8.0.0` to pin to v8 releases, so that we can gradually adopt new features in new minor releases.
> - We add PHP 8 support via the (insert appropriate Travis-CI build target here) and configure it using ... 

etc. etc. etc.

File: meetings/agenda.md (hunk 1 above)
Comment (boesing) [reply meetings/agenda.md:13]: tbh. even if 7.2 is alive, people can stick to current versions and we probably will backport security fixes if needed. 
I would also propose supporting only actively developed php versions, which will be, at the end of the year, 7.4 and 8.0
If people can stick with legacy php versions, they dont need new features either.

**Comment by boesing** [comment]
> Tell us how YOU think it should be done.

I dont have as much experience as you guys have. Thats why I usually dont propose stuff as I think there are peeps who are more into this than I am. Its my first open source Major - didnt you guys took learnings from 5 to 7? 😅

**Comment by Ocramius** [comment]
> didnt you guys took learnings from 5 to 7? sweat_smile

Yes: mostly that everything breaks during the last few RCs :P 

In practice, we either run tests against latest `8.0` and they are green on `8.0.0 GA` too, or we lied :D

**Comment by boesing** [comment]
> In practice, we either run tests against latest 8.0 and they are green on 8.0.0 GA too, or we lied :D

Sure thing. 
But what I've found out was, that supporting `^8.0` would be too generic as something might break with `8.1`. 
Thats why I want to know if adding `~8.0.0` would be acceptable or not.
There was a huge discussion that symfony supports `>=7.x` in most packages.
Most people were fine with that change as it does not block migrating to PHP 8, others said its like removing the PHP dependency at all, and so on.
Thats why I like to get feedback on this topic as I am too new to that whole "new php major version support" thing.

**Pull request closed without merging**